app/gui/controller/double-representation/src/definition.rs

```diff
@@ -413,6 +441,12 @@ pub trait DefinitionProvider {
         let indent = self.indent();
         DefinitionIterator { iterator, scope_kind, indent }
     }
+
+    /// Returns an iterator iterating recursively over definitions provided by this provider and
+    /// their nested definitions (as [`ChildDefinition`] is also a provider).
+    fn recursive_def_iter(&self) -> RecursiveDefinitionIterator {
```

Comment:
Would it be enough to have the return type `impl Iterator<Item=ChildDefinition>`?

Reply:
Cannot return `impl` in traits definition.